assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -566,12 +566,12 @@ public ActionReturn getGrades(Map<String, Object> params) {
         }));
 
         Map<String, GraderUser> students = new HashMap<>();
-        try {
-            for (String studentId : site.getUsersIsAllowed(SECURE_ADD_ASSIGNMENT_SUBMISSION)) {
+        for (String studentId : site.getUsersIsAllowed(SECURE_ADD_ASSIGNMENT_SUBMISSION)) {
+            try {
                 students.put(studentId, new GraderUser(userDirectoryService.getUser(studentId)));
```

Comment:
make this a batch call up above instead;

userDirectoryService.getUsers( Set or List) and that will return only valid users 

this will make one call to institutional LDAP instead of XX individual calls

Reply:
Cool, will do.

Reply:
Done.